File: testing_denoise/Bilateral.py
import numpy as np
import cv2
from matplotlib import pyplot as plt
import os

dir_name = "./testing_image/"
images = os.listdir(dir_name)

f, a = plt.subplots(3, 4, figsize=(7,7))
var = [50,100,200]
title = "d"
f.suptitle(title)
for i, imname in enumerate(images):
  showimg = []
  img = cv2.imread(dir_name+imname)
  showimg.append(img)

  for j in var:
    dst = cv2.bilateralFilter(img)
    showimg.append(dst)

  for j in range(len(var)+1):
    if j!=0:
      a[j//4][j%4].set_title(var[j-1])
    a[j//4][j%4].imshow(showimg[j], cmap="gray")
    a[j//4][j%4].set_xticks(()); a[j//4][j%4].set_yticks(());

  plt.show()
  f.savefig("./result/find_interval/"+imname+"Bilateral_"+title+".png")

File: testing_denoise/Bilateral.m
delete(gcp)
parpool(4);

dirname = './testing_image/';
savedirname = './result/find_interval/';
MyFolderInfo = dir(dirname);

parfor j=3:length(MyFolderInfo)
    var=[10^-8,10^-7,10^-6,10^-5,10^-4,10^-3,10^-2,10^-1,10^0,10^1,10^2];
    %var=[2^-5,2^-4,2^-3,2^-2,2^-1,2^0,2^1,2^2,2^3,2^4,2^5];
    %var=[1,3,5,7,9,11,13,17,33,65,129]
    showimg=zeros(256,256,12);
    img = imread(strcat(dirname, MyFolderInfo(j).name));
    img = im2double(img);
    showimg(:,:,1)=img;
    
    for i = 1:11
        dst = imbilatfilt(img,var(i));
        %dst = imbilatfilt(img,0.01,var(i));
        %dst = imbilatfilt(img,0.01,1,'NeighborhoodSize',var(i));
        showimg(:,:,i+1)=dst;
    end
    
    figure;
    title('degreeOfSmoothing');
    for i = 1:12
        subplot(3,4,i), imshow(showimg(:,:,i));
        
        if i==1
           title('original');
        else
           title(num2str(var(i-1)));
        end
    end
    tightfig;
    saveas(gcf,strcat(savedirname,MyFolderInfo(j).name,'_Bilateral_degreeOfSmoothing.png'));
end